### DISCUSSION TITLE
query date after x amount

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2595] [Q&A]
I have a table with a column `createdAt` and I want to get all the rows that are older than, say, 10 days after the creation date. How would the query be?
Something like
```dart
select(myTable).where((tbl) => tbl.createdAt.isBiggerThan(tbl.createdAt + Duration(days: 10)));
```

**Comment by simolus3** [comment]
If you want to find all rows that have been created at least ten days ago, I would use

```dart
select(myTable).where((tbl) => currentDateAndTime.isBiggerThan(tbl.createdAt.modify(const DateTimeModifier.days(10))));
```

Using `+` will add the second-equivalent of 10 days, which can sometimes lead to undesired results because not every calendar day is 24h long. `modify` is a sqlite3-specific API that tries to add the days directly. If you need to support MariaDB, using `currentDateAndTime.isBiggerThan(tbl.createdAt + Duration(days: 10))` is probably the best we can do right now.

**Reply by BananaMasterz** [reply]
the example I gave isn't functional since the syntax isn't correct, I just typed whatever kinda since I couldn't figure the syntax of doing something like that, as you guessed it's for mariadb and the `.modify` doesn't work, currently im doing this 
```dart
(tbl) => tbl.createdAt.isSmallerThanValue(DateTime.now().subtract(Duration(days: 10))))
```
but it's confusing a bit

**Reply by simolus3** [reply]
Yeah that should work as well, but you can also use `currentDateAndTime` as an expression which evaluates to the current time in the database server.

**Reply by BananaMasterz** [reply]
Not sure how I would do that, can I subtract from using that?

**Reply by simolus3** [reply]
`currentDateAndTime` evaluates to an `Expression<DateTime>`, for which drift defines [the plus and minus operator](https://drift.simonbinder.eu/api/drift/datetimeexpressions), yes.
